### PR TITLE
Optout ld lib  HPC-5236

### PR DESCRIPTION
**Pull request by JensTimmerman**

### CODE REVIEW

File: SOURCES/environment-modules-optional_ld.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+--- modules-3.2.10.orig/init/bash.in	2012-10-25 21:33:34.000000000 +0200
++++ modules-3.2.10/init/bash.in	2015-06-03 03:26:16.512608933 +0200
+@@ -7,8 +7,8 @@
+ @VERSIONING@fi
+ @VERSIONING@export MODULE_VERSION_STACK
+ 
+-@VERSIONING@module() { eval `@BASEPREFIX@/Modules/$MODULE_VERSION/bin/modulecmd bash $*`; }
+-@NOTVERSIONING@module() { eval `@bindir@/modulecmd bash $*`; }
++@VERSIONING@module() { eval `@BASEPREFIX@/Modules/$MODULE_VERSION/bin/modulecmd bash $*`; if [ "$FORWARD_COMPAT_LD_PRELOAD" ]; then export LD_PRELOAD=''; fi; if [ "$FORWARD_COMPAT_LD_LIBRARY_PATH" ]; then export LD_LIBRARY_PATH="/lib64:/usr/lib64:$(echo $LD_LIBRARY_PATH | LD_LIBRARY_PATH="/lib64" /bin/sed 's/:\/lib64:\/usr\/lib64:/:/g')"; fi;}
++@NOTVERSIONING@module() { eval `@bindir@/modulecmd bash $*`; if [ "$FORWARD_COMPAT_LD_PRELOAD" ]; then export LD_PRELOAD=''; fi; if [ "$FORWARD_COMPAT_LD_LIBRARY_PATH" ]; then export LD_LIBRARY_PATH="/lib64:/usr/lib64:$(echo $LD_LIBRARY_PATH | LD_LIBRARY_PATH="/lib64" /bin/sed 's/:\/lib64:\/usr\/lib64:/:/g')"; fi;}
```
Comment (boegel) [inline SOURCES/environment-modules-optional_ld.patch]: where's the stuff we had to track module usage?!

Comment (JensTimmerman) [inline SOURCES/environment-modules-optional_ld.patch]: I didn't touch that, so still in https://github.com/hpcugent/environment-modules/blob/master/SOURCES/modulecmd

Comment (boegel) [inline SOURCES/environment-modules-optional_ld.patch]: Ah, ok, we're hijacking the 'binary' for that part, ok, sorry.

File: SOURCES/environment-modules-optional_ld.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+--- modules-3.2.10.orig/init/bash.in	2012-10-25 21:33:34.000000000 +0200
++++ modules-3.2.10/init/bash.in	2015-06-03 03:26:16.512608933 +0200
+@@ -7,8 +7,8 @@
+ @VERSIONING@fi
+ @VERSIONING@export MODULE_VERSION_STACK
+
+-@VERSIONING@module() { eval `@BASEPREFIX@/Modules/$MODULE_VERSION/bin/modulecmd bash $*`; }
+-@NOTVERSIONING@module() { eval `@bindir@/modulecmd bash $*`; }
++@VERSIONING@module() { eval `@BASEPREFIX@/Modules/$MODULE_VERSION/bin/modulecmd bash $*`; if [ "$FORWARD_COMPAT_LD_PRELOAD" ] && [ "$VSC_OS_LOCAL" != "SL6" ]; then export LD_PRELOAD=''; fi; if [ "$FORWARD_COMPAT_LD_LIBRARY_PATH" ] && [ "$VSC_OS_LOCAL" != "SL6" ]; then export LD_LIBRARY_PATH="/lib64:/usr/lib64:$(echo $LD_LIBRARY_PATH | LD_LIBRARY_PATH="/lib64" /bin/sed 's/:\/lib64:\/usr\/lib64:/:/g')"; fi;}
++@NOTVERSIONING@module() { eval `@bindir@/modulecmd bash $*`; if [ "$FORWARD_COMPAT_LD_PRELOAD" ] && [ "$VSC_OS_LOCAL" != "SL6" ]; then export LD_PRELOAD=''; fi; if [ "$FORWARD_COMPAT_LD_LIBRARY_PATH" ] && [ "$VSC_OS_LOCAL" != "SL6" ]; then export LD_LIBRARY_PATH="/lib64:/usr/lib64:$(echo $LD_LIBRARY_PATH | LD_LIBRARY_PATH="/lib64" /bin/sed 's/:\/lib64:\/usr\/lib64:/:/g')"; fi;}
```
Comment (boegel) [inline SOURCES/environment-modules-optional_ld.patch:10]: please add some comments explaining the hackery in here, so we are reminded of why this is needed when checking `type module`

why is this patch so horrible btw (w.r.t. long lines), i.e. why doesn't it define the module function multiline?

Comment (boegel) [inline SOURCES/environment-modules-optional_ld.patch:10]: also, guess what's going to happen if we update the login nodes from SL6 to whatever is next...

Comment (JensTimmerman) [inline SOURCES/environment-modules-optional_ld.patch:10]: see HPC-5236 for a discussion on that

Comment (boegel) [inline SOURCES/environment-modules-optional_ld.patch:10]: on what aspect exactly?

Comment (JensTimmerman) [inline SOURCES/environment-modules-optional_ld.patch:10]: all of them?

**Comment by JensTimmerman** [comment]
this rpm has been built and is deployed on our systems, so better to have this code in master here so we can find it when we need it.

**Pull request merged**